native/src/main/java/io/ballerina/persist/nodegenerator/SyntaxTreeGenerator.java

```diff
@@ -0,0 +1,201 @@
+import static io.ballerina.persist.PersistToolsConstants.KEY_PASSWORD;
+import static io.ballerina.persist.PersistToolsConstants.KEY_PORT;
+import static io.ballerina.persist.PersistToolsConstants.KEY_USER;
+
+
+/**
+ * Class to create syntax tree for Config.toml.
+ */
+public class SyntaxTreeGenerator {
+    private static final String[] nodeMap = {KEY_HOST, KEY_PORT, KEY_USER, KEY_DATABASE, KEY_PASSWORD};
+    private static final String[] defaultValues = {DEFAULT_HOST, DEFAULT_PORT, DEFAULT_USER, DEFAULT_DATABASE,
+            DEFAULT_DATABASE};
+
+    /**
+     * Method to create a new Config.toml file with database configurations.
+     */
+    public static SyntaxTree createToml(String name) {
+        NodeList moduleMembers = AbstractNodeFactory.createEmptyNodeList();
+        moduleMembers = moduleMembers.add(SampleNodeGenerator.createTable(name, null));
+        moduleMembers = populateNodeList(moduleMembers);
+        Token eofToken = AbstractNodeFactory.createIdentifierToken("");
+        DocumentNode documentNode = NodeFactory.createDocumentNode(moduleMembers, eofToken);
+        TextDocument textDocument = TextDocuments.from(documentNode.toSourceCode());
+        return SyntaxTree.from(textDocument);
+    }
+
+    /**
+     * Method to update the Config.toml with database configurations.
+     */
+    public static SyntaxTree updateToml(Path configPath, String name) throws IOException {
+
+        ArrayList<String> existingNodes = new ArrayList<>();
+        NodeList<DocumentMemberDeclarationNode> moduleMembers = AbstractNodeFactory.createEmptyNodeList();
+        Path fileNamePath = configPath.getFileName();
+        TextDocument configDocument = TextDocuments.from(Files.readString(configPath));
+        SyntaxTree syntaxTree = SyntaxTree.from(configDocument, fileNamePath.toString());
+        DocumentNode rootNote = syntaxTree.rootNode();
+        NodeList nodeList = rootNote.members();
+
+        for (Object member : nodeList) {
+            if (member instanceof KeyValueNode) {
+                moduleMembers = moduleMembers.add((DocumentMemberDeclarationNode) member);
+            } else if (member instanceof TableNode) {
+                TableNode node = (TableNode) member;
+                if (node.identifier().toString().trim().equals(name)) {
+                    NodeList subNodeList = node.fields();
+                    moduleMembers = moduleMembers.add(SampleNodeGenerator.createTable(name, null));
+                    for (Object subMember : subNodeList) {
+                        if (!isDatabaseConfigurationEntry(subMember.identifier())) {
+                            moduleMembers = moduleMembers.add((DocumentMemberDeclarationNode) subMember);
+                        } else {
+                            if (subNode.identifier().toString().trim().equals(KEY_PORT)) {
+                                existingNodes.add(subNode.identifier().toString().trim());
```

Comment:
This is duplicated in both if and else blocks.

Reply:
Two blocks are used since port is a `numericalKV` while others are `StringKV`

Reply:
`existingNodes.add(subNode.identifier().toString().trim());` is duplicated in both blocks. Better to take it out.

Reply:
It was updated in [53b94a5](https://github.com/ballerina-platform/persist-tools/pull/9/commits/53b94a557e6bdb6ac1800b8eb683bda757695cc5)